examples/kitty/kitty.mk

```diff
@@ -164,16 +160,45 @@ nfs.elf: nfs/nfs.a libnfs/lib/libnfs.a musllibc/lib/libc.a
 
 ${IMAGES}: libsddf_util_debug.a
 
-%.o: %.c
+%.o: %.c ${SDDF}/include
 	${CC} ${CFLAGS} -c -o $@ $<
 
 $(IMAGE_FILE) $(REPORT_FILE): $(IMAGES) ${KITTY_DIR}/kitty.system
 	$(MICROKIT_TOOL) ${KITTY_DIR}/kitty.system --search-path $(BUILD_DIR) --board $(MICROKIT_BOARD) --config $(MICROKIT_CONFIG) -o $(IMAGE_FILE) -r $(REPORT_FILE)
 
 FORCE:
 
-%.elf: %.o
-	${LD} ${LDFLAGS} -o $@ $< ${LIBS}
-
-mpy-cross: FORCE
+mpy-cross: FORCE ${LIONSOS}/dep/micropython/mpy-cross
 	${MAKE} -C ${LIONSOS}/dep/micropython/mpy-cross BUILD=$(abspath ./mpy_cross)
+
+# If you want to use your own VM for the graphics driver
+# then change these lines or just make sure you've already put
+# linux and rootfs.cpio.gz into the Build directory
+${LINUX}:
+	curl -L ${LIONSOS_DOWNLOADS}/$(KITTY_GRAPHICS_VM_LINUX) -o $@
+
+${INITRD}:
+	curl -L ${LIONSOS_DOWNLOADS}/$(KITTY_GRAPHICS_VM_ROOTFS) -o $@
+
+# Because we include the *.mk files they are an implicit dependency
+# for the build.  These rules instantiate the submodules that
+# include the makefiles.
+${LIBVMM_DIR}/vmm.mk:
```

Comment:
These check for the *existence* of these files not if the files have changed right? E.g if you pulled a new version of libvmm that changed `vmm.mk` would it update the submodule?